cishouseholds/pipeline/reporting.py

```diff
@@ -51,3 +53,25 @@ def unmatching_antibody_to_swab_viceversa(swab_df, antibody_df, column_list):
 
     df = union_multiple_tables(tables=[unmatched_swab_df, unmatched_antibody_df])
     return df
+
+
+def generate_error_table(table_name: str, error_priority_map: dict, validation_failure_flag_column: str):
+    df = extract_from_table(table_name)
+    df = df.withColumn("order", F.col("validation_check_failures"))
+    df = update_column_values_from_map(df, "order", error_priority_map, default_value=9999).orderBy("order")
+    head = df.select("run_id").orderBy(F.col("run_id").desc()).distinct().head(2)
```

Comment:
There's a risk that this picks up data that isn't from the current run (e.g. if the validation stage has failed for one or more runs). So could we instead use `get_run_id()` amd `get_run_id() -1` to select the records please?

It would be good to remove the if statement here, so that it always displays two count columns. This will make it clear when values were not available for one of the runs. The join should still work, even if one of the filter statements returns no rows